### ISSUE TITLE
Texture Loading Issue when Blender Language is Set to Japanese

### ISSUE THREAD

**Issue by TomXV** [NSACloud/RE-Mesh-Editor#7]
Hello, and thank you for this useful addon!

I encountered an issue where textures were not loading correctly. Despite having the "Material Output" node in place, it was not detected, and the console displayed an error indicating that the node was missing.

I tried various troubleshooting steps, including changing file paths, but the issue persisted.

Then, as a last attempt, I changed Blender’s language setting from Japanese to English. To my surprise, this resolved the issue, and the addon functioned correctly.

## **Question**
Is this a bug or an expected limitation? If this behavior is intentional, could you please consider adding a note in the README.md file to inform users that the addon works best when Blender’s language is set to English?

---

Thank you very much for your time and support!

**Comment by NSACloud** [comment]
I'll look at fixing the issue later today. In the addon preferences, is your texture cache path set to the default location (inside the appdata/addon folder)?

If you change the texture cache path to another location, do textures load when the language is set to Japanese?

I'm assuming what's happening is that Blender is localizing the file path into Japanese and that might be what's causing the problem.

**Comment by TomXV** [comment]
Thank you for your response.

In answer to your questions:

1. 
> **In the addon preferences, is your texture cache path set to the default location (inside the appdata/addon folder)?**

→ Yes, it is set to the default location.

2. 
> **If you change the texture cache path to another location, do textures load when the language is set to Japanese?**  

→ I tried changing the texture cache path to a different location, but the textures still didn’t load. Instead, I encountered the 

following error:

   ```powershell
   KeyError: 'bpy_prop_collection[key]: key "Material Output" not found'
   繧ｨ繝ｩ繝ｼ: Python: Traceback (most recent call last):
     File "C:\Users\user\AppData\Roaming\Blender Foundation\Blender\4.2\scripts\addons\RE-Mesh-Editor-main\modules\mdf\re_mdf_operators.py", line 108, in execute
       importMDF(mdfFile, meshMaterialDict, bpy.context.scene.re_mdf_toolpanel.loadUnusedTextures, bpy.context.scene.re_mdf_toolpanel.loadUnusedProps, bpy.context.scene.re_mdf_toolpanel.useBackfaceCulling, bpy.context.scene.re_mdf_toolpanel.reloadCachedTextures, chunkPath=modDir, gameName=bpy.context.scene.re_mdf_toolpanel.activeGame, arrangeNodes=True)
     File "C:\Users\user\AppData\Roaming\Blender Foundation\Blender\4.2\scripts\addons\RE-Mesh-Editor-main\modules\mdf\blender_re_mesh_mdf.py", line 1366, in importMDF
       links.new(matInfo["currentShaderOutput"], nodes["Material Output"].inputs["Surface"])
                                              ~~~~~^^^^^^^^^^^^^^^^^^^
   KeyError: 'bpy_prop_collection[key]: key "Material Output" not found'
   ```

It seems that Blender might be localizing the "Material Output" node name when the language is set to Japanese, changing it to “マテリアル出力.” This could be causing the mismatch, as the node name doesn’t align with what the addon expects.

Could you please check on that point?

**Comment by TomXV** [comment]
Okay, understood.

And, I noticed something that might be related to the issue. I tried manually changing the node name to “Material Output” and then clicking “Apply Active MDF”, but it keeps switching back to Japanese. I wasn’t sure if this was caused by the addon or Blender itself, so I also tested with Chinese, and the same issue occurred.

It seems that when the addon tries to overwrite the node, Blender may automatically translate the node name according to the language setting. So, I believe this may be a Blender specification.

After reviewing the bpy library documentation, I found that using internal identifiers like bl_idname instead of node names might be a potential solution to avoid this issue.

Here’s the reference for bl_idname in Blender’s documentation:
https://docs.blender.org/api/current/bpy.types.Node.html

I hope this information is helpful. Currently, it seems the addon only works in English due to localization limitations, but I look forward to any future improvements for multilingual support🚀
